rtos/TARGET_CORTEX/mbed_boot.c

```diff
@@ -187,7 +187,11 @@ osThreadAttr_t _main_thread_attr;
 #ifndef MBED_CONF_APP_MAIN_STACK_SIZE
 #define MBED_CONF_APP_MAIN_STACK_SIZE 4096
 #endif
+#if defined(__CORTEX_A9)
+MBED_ALIGN(8) char _main_stack[MBED_CONF_APP_MAIN_STACK_SIZE*4];
```

Comment:
why we cant use the default one (*4 here)? This is breaking this config. As a user would set it to 10k for instance, actually would get 40k.

Reply:
This change refer in the processing of previous mbed-os version.
Reference is [here](https://github.com/ARMmbed/mbed-os/blob/mbed-os-5.4.7/rtos/rtx/TARGET_CORTEX_A/RTX_CM_lib.h#L423).
I think Cortex-A in general will require more stack space more than Cortex-M.

Reply:
As we can see, some changes like this would be good to have as a separate commit with description to understand  the reason for this change.

The default value should be rather changed. It is set on the line 188 in this file. As we would need to do *4 for any line that uses it in this file.

> I think Cortex-A in general will require more stack space more than Cortex-M.

4k by default would not be sufficient to run mbed-os examples ? If we agree that Cortex-A should have 4x more stack by default than cortex-m, then we would need:

```
#if cortex-a
// default stack is 16k
#else
// default stack is 4k
#endif
```

@bulislaw Please check

Reply:
We decided to set it to 4K by default for all platforms and make it configurable for applications. Overwriting it silently like that is confusing. Why do we need to do that? Is 4K not enought to run basic Mbed Apps and examples?

Reply:
In case of [this test](https://github.com/ARMmbed/mbed-os/blob/master/TESTS/mbedmicro-rtos-mbed/malloc/main.cpp), Executing with `#define THREAD_STACK_SIZE   256` in Cortex-A, program will abort because of the lack of stack.
As a result of check, in Cortex-A, program will work properly by `#define THREAD_STACK_SIZE    512` at least.
(Now the value of macro I committed is [2048](https://github.com/ARMmbed/mbed-os/pull/5530/commits/aa328244abad0ec0d08b7bf5f0755551b705ec04#diff-03297e82aaace147c0ac9db765ba2780R27), I will change to 512)

This is why, Cortex-A needs more stack size than Cortex-M.
Since this value(16K) may be too large, how about change the value of this from 16K to 8K by following the above size?
If there are no problems with my propose, I will change my commit soon.
Not to mention the change of `THREAD_STACK_SIZE` in the above test.

Reply:
@0xc0170 @bulislaw
I will appreciate if you reply this promptly.

Reply:
I would leave the `MBED_CONF_APP_MAIN_STACK_SIZE` as it is for all platforms. The question above was not answered, if 4k is not sufficient for mbed OS app. Did you have any failures with it - tests or examples?
This is application specific, it is intended to be tuned by application.

Reply:
@0xc0170 
Thank you for your commnets.
I understood the specification of `MBED_CONF_APP_MAIN_STACK_SIZE`.
To be honest, I was worried about the lack of the stack by any chance, there was no problem for mbed test in case of 4K. 
Therefore, I will modify to revert the value of macro to 4K soon.

Reply:
I committed.